python/zqd/zqd.py

```diff
@@ -0,0 +1,104 @@
+import datetime
+import decimal
+import ipaddress
+import json
+import requests
+
+
+DEFAULT_BASE_URL = 'http://127.0.0.1:9867'
+
+
+class Client():
+
+    def __init__(self, base_url=DEFAULT_BASE_URL):
+        self.base_url = base_url
+        self.session = requests.Session()
+
+    def ast(self, zql):
+        r = self.session.post(self.base_url + "/ast", json={'zql': zql})
+        r.raise_for_status()
+        return r.json()
+
+    def search(self, space_name, zql):
+        aliases, schemas = {}, {}
+        for raw in self.search_raw(space_name, zql):
+            if raw['type'] == 'SearchRecords':
+                for rec in raw['records']:
+                    if 'aliases' in rec:
```

Comment:
It seems like the the process of parsing zqd search results into a stream of zjson JSON objects should be separate from the thing that parses a stream of zjson JSON objects into a stream of native python objects.  This way, for example you could have a python program that reads a file that has zjson in and parses the zjson into the python objects.  This should also help people reading the code to understand the model here more easily.

Reply:
This would also make tests easier to write as this subsystem evolves.

Reply:
@mccanne: Is this what you had in mind?

```python
class Client():
    ...
    def search(self, space_name, zql):
        raw = self.search_raw(space_name, zql)
        zjson = take_zjson(raw)
        return decode_zjson(zjson)


def take_zjson(raw):
    for obj in raw:
        if obj['type'] == 'SearchRecords':
            for zjson in obj['records']:
                yield zjson


def decode_zjson(zjson):
    aliases, schemas = {}, {}
    for obj in zjson:
        if 'aliases' in obj:
            for a in obj['aliases']:
                aliases[a['name']] = a['type']
        if 'schema' in obj:
            schemas[obj['id']] = obj['schema']
        yield _to_native(aliases, schemas[obj['id']], obj['values'])
```

Reply:
I think so.

Reply:
Done in cf76c327ba99e196236e177c0f9a259c7e64c135.